src/components/single-product-card/SingleProductCard.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+import { useCart, useAuth, useWishlist } from "../../context/index";
+import { useNavigate } from "react-router-dom";
+import "./single-product-card.css";
+
+function SingleProductCard({ product }) {
+  const [isLoading, setIsLoading] = useState({
+    cart: false,
+    wishlist: false,
+  });
+
+  const navigate = useNavigate();
+
+  const { cartItems, addToCart } = useCart();
+  const { wishlistItems, addToWishlist, removeFromWishlist } = useWishlist();
+  const isInWishlist = wishlistItems.find((item) => item._id === product?._id);
```

Comment:
You're using `isInWishlist` in multiple places (here and in `ProductCard.jsx`), so you can export it and make it a helper function.

Reply:
Thanks for the suggestion, I will do that while refactoting.